From: Bhavna Sarathy <bnagendr@redhat.com>
Date: Thu, 4 Feb 2010 15:58:15 -0500
Subject: [x86] fix Add AMD Node ID MSR support
Message-id: <4B6AEE97.9090701@redhat.com>
Patchwork-id: 23127
O-Subject: Re: [RHEL6.0 PATCH] Add AMD Node ID MSR support
Bugzilla: 557540
RH-Acked-by: Andrew Jones <drjones@redhat.com>

> Resolves 557540
>
> AMD has added a NodeID MSR to the BIOS which avoids accessing the PCI
> configuration space to access the Node ID.  This MSR will be supported
> by all G34 multi-node production systems.
>
> Upstream:
> http://lkml.org/lkml/2009/10/16/111
>
> Please review and ACK
Good catch, here's the fix-up.   Sanity checks done.  Please review and ACK.

Signed-off-by: Bhavna Sarathy <bhavna.sarathy@amd.com>
Signed-off-by: Aristeu Rozanski <arozansk@redhat.com>

diff --git a/arch/x86/include/asm/msr-index.h b/arch/x86/include/asm/msr-index.h
index 4ffe09b..9d976a7 100644
--- a/arch/x86/include/asm/msr-index.h
+++ b/arch/x86/include/asm/msr-index.h
@@ -123,6 +123,7 @@
 #define FAM10H_MMIO_CONF_BUSRANGE_SHIFT 2
 #define FAM10H_MMIO_CONF_BASE_MASK	0xfffffff
 #define FAM10H_MMIO_CONF_BASE_SHIFT	20
+#define MSR_FAM10H_NODE_ID		0xc001100c
 
 /* K8 MSRs */
 #define MSR_K8_TOP_MEM1			0xc001001a
diff --git a/arch/x86/kernel/cpu/amd.c b/arch/x86/kernel/cpu/amd.c
index 7476578..e485825 100644
--- a/arch/x86/kernel/cpu/amd.c
+++ b/arch/x86/kernel/cpu/amd.c
@@ -270,7 +270,7 @@ static void __cpuinit amd_fixup_dcm(struct cpuinfo_x86 *c)
 	if (cpu_has(c, X86_FEATURE_AMD_DCM))
 		return;
 
-	rdmsrl(0xc001100c, value);
+	rdmsrl(MSR_FAM10H_NODE_ID, value);
 
 	nodes = ((value >> 3) & 7) + 1;
 	if (nodes == 1)
